### ISSUE TITLE
Shuffle Features

### ISSUE THREAD

**Issue by idksomethingedgy** [Soundnode/soundnode-app#856]
The shuffle feature, whether it be in a playlist or in liked videos doesn't work correctly.

- Shuffling is limited to a certain amount of your songs, meaning it will replay the same songs without pulling new ones

- Shuffling can cause the same song to replay after 2-3 new songs

- Going to the previous song while shuffling picks a completely new song, It would be nice to have it go to the previous song

- http://i.imgur.com/qtEnqqY.png  //  This is limited to a certain amount of songs (this could relate to the shuffle issue where it only cycles through certain songs) 

This is for any playlist, whether that be your own playlists, likes, or an album someone has uploaded.

**Comment by weblancaster** [comment]
Yes the shuffle algorithm need to be improved, what Soundnode have right now is just a placeholder and not really a shuffle.
